File: src/main/resources/static/js/app/ReceitaService.js
'use strict'

angular.module('opusreceitas.services', []).factory('ReceitaService',
		[ "$http", "CONSTANTS", function($http, CONSTANTS) {
			var service = {};
			service.getReceitaById = function(id) {
				var url = CONSTANTS.getReceitaById + id;
				return $http.get(url);
			}
			service.criarReceita = function(receitaDto) {
				return $http.post(CONSTANTS.criarReceita, receitaDto);
			}
			service.listarIngredientes = function() {
				return $http.get(CONSTANTS.getAllIngredientes);
			}
			service.listarReceitas= function() {
				return $http.get(CONSTANTS.getAllReceitas);
			}
			return service;
		} ]);

File: src/main/java/com/opus/receitas/service/ReceitaService.java
package com.opus.receitas.service;

import java.util.List;

import com.opus.receitas.dto.IngredientesDto;
import com.opus.receitas.dto.ReceitaDto;

public interface ReceitaService {
    ReceitaDto getReceitaById(Integer id);
    void saveReceita(ReceitaDto receitaDto);
    List<ReceitaDto> getAllReceitas();
    List<IngredientesDto> getAllIngredientes();
}
